client/src/components/Appointment.js

```diff
@@ -77,18 +141,18 @@ return (
       <input type="date" id="date" name="date" value={date} onChange={handleDateChange}></input>
     </div>
     <div id="select-time">
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={9}>9:00 AM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={10}>10:00 AM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={11}>11:00 AM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={12}>12:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={13}>1:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={14}>2:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={15}>3:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={16}>4:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={17}>5:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={18}>6:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={19}>7:00 PM</button>
-      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn} onClick={handleTimeChange} value={20}>8:00 PM</button>
+      <button className="btn btn-lg btn-primary" disabled={disableTimeBtn || blockedTimeBtn[0].blocked} onClick={handleTimeChange} value={9}>9:00 AM</button>
```

Comment:
I believe that the react way to do this is using a function that returns an array of button components.

Reply:
That's not needed in this case, as this is not being rendered from the data pulled from the database, they will always be there, the only thing that changes is the state between disabled being true or false depending on the state.